audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -221,20 +249,84 @@ void testRestEndpointForGoodSuccessMessage() {
 
     @Test
     @DirtiesContext
-    void testRestEndpointForBadSuccessMessage() {
+    void testRestEndpointForBadSuccessMessage() throws InterruptedException {
         // Pass a success message to 'success' rest endpoint
         // When - we POST to the rest endpoint
         Map<String, List<String>> headers = Collections.singletonMap(Token.HEADER, Collections.singletonList(ContractTestData.REQUEST_TOKEN));
         HttpEntity<AuditSuccessMessage> entity = new HttpEntity<>(ContractTestData.BAD_SUCCESS_REQUEST_OBJ, new LinkedMultiValueMap<>(headers));
         ResponseEntity<Void> response = restTemplate.postForEntity("/api/success", entity, Void.class);
 
+        TimeUnit.SECONDS.sleep(1);
+
         // Then - check the REST request was accepted
         assertThat(response.getStatusCode()).isEqualTo(HttpStatus.ACCEPTED);
 
         // Then - check the audit service has invoked the audit method
         Mockito.verify(auditService, Mockito.timeout(3000).times(0)).audit(anyString(), any());
     }
 
+    @Test
+    @DirtiesContext
+    void testFailedErrorDeserialization() throws InterruptedException {
+        // Add a message to the 'error' topic
+        // The ContractTestData.REQUEST_TOKEN maps to partition 0 of [0, 1, 2], so the akka-test yaml connects the consumer to only partition 0
+        final Stream<ProducerRecord<String, JsonNode>> requests = ContractTestData.BAD_ERROR_MESSAGE_NODE_FACTORY.get().limit(1L);
+
+        // When - we write to the input
+        ProducerSettings<String, JsonNode> producerSettings = ProducerSettings
+                .create(akkaActorSystem, new StringSerializer(), new RequestSerializer())
+                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
+
+        Source.fromJavaStream(() -> requests)
+                .runWith(Producer.plainSink(producerSettings), akkaMaterializer);
+
+        TimeUnit.SECONDS.sleep(2);
+
+        // Then check an "Error-..." file has been created
+        Set<File> expectedFileSet = Arrays.stream(new File(".").listFiles())
+                .filter(file -> file.getName().startsWith("Error"))
+                .collect(Collectors.toSet());
+        Set<File> actualFileSet = Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
+                .filter(file -> file.getName().startsWith("Error"))
+                .collect(Collectors.toSet());
```

Comment:
Ditto, `assert .getErrorDirectory() == "."` is not going to provide an appropriate test case

Reply:
Done here as well